core-rust/state-manager/src/state_manager.rs

```diff
@@ -712,123 +711,132 @@ where
 
         // Whilst we could validate intent hash duplicates here, these are checked by validators on prepare already,
```

Comment:
By the way - the panic on line 704 - I assume we check in Java that the sync response contains transactions before sending it to commit in Rust?

Reply:
Yes: https://github.com/radixdlt/babylon-node/blob/524ab532b6f65cf615c4460a1fc87548f8896228/core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java#L329-L334

(to be specific: we only _use_ the sync response to compute the actual "ledger extension" that we pass to Rust commit; and the computation requires basically that "there must be a transaction with state version greater than current ledger's version", which means that list of transactions cannot be empty)